src/Servers/IIS/IIS/src/NativeMethods.cs

```diff
@@ -56,14 +49,14 @@ public enum REQUEST_NOTIFICATION_STATUS
         private static extern void http_indicate_completion(NativeSafeHandle pInProcessHandler, REQUEST_NOTIFICATION_STATUS notificationStatus);
 
         [DllImport(AspNetCoreModuleDll)]
-        private static extern int register_callbacks(NativeSafeHandle pInProcessApplication,
-            PFN_REQUEST_HANDLER requestCallback,
-            PFN_SHUTDOWN_HANDLER shutdownCallback,
-            PFN_DISCONNECT_HANDLER disconnectCallback,
-            PFN_ASYNC_COMPLETION asyncCallback,
-            PFN_REQUESTS_DRAINED_HANDLER requestsDrainedHandler,
-            IntPtr pvRequestContext,
-            IntPtr pvShutdownContext);
+        private unsafe static extern int register_callbacks(NativeSafeHandle pInProcessApplication,
+            delegate* unmanaged[Cdecl]<nint, nint, REQUEST_NOTIFICATION_STATUS> requestCallback,
```

Comment:
The original delegates had default unmanaged calling convention. Explicitly specifying `Cdecl` here is changing behavior. 

I think this should be just `delegate* unmanaged<nint, nint, REQUEST_NOTIFICATION_STATUS>`. Similar for all other places.

Reply:
I was confused about this. Will change it back.